### PR TITLE
Quadruna RSM Setup

### PR DESCRIPTION
**Pull request by shayana18**
### Resolved Issues

- Implements [SW-90](https://ubcformulaelectric.atlassian.net/jira/software/projects/SW/boards/20?selectedIssue=SW-90)

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [ ] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [ ] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

[SW-90]: https://ubcformulaelectric.atlassian.net/browse/SW-90?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: firmware/quadruna/RSM/src/io/io_coolant.c [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+{
+    // TODO calculate the pressure
+    return hw_adc_getVoltage(ADC1_IN12_COOLANT_PRESSURE_1);
+}
+
+float io_coolant_getPressureB(void)
+{
+    // TODO calculate the pressure
+    return hw_adc_getVoltage(ADC1_IN11_COOLANT_PRESSURE_2);
+}
```
Comment (Lucien950) [inline firmware/quadruna/RSM/src/io/io_coolant.c:98]: pressure transfer function

File: firmware/quadruna/RSM/src/io/io_brakeLight.c [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#include "io_led.h"
+#include "app_globals.h"
+#include "app_canRx.h"
+
+void brakeLightEnable()
```
[the remark this answers is not in the document]
Comment (Lucien950) [reply firmware/quadruna/RSM/src/io/io_brakeLight.c:5]: also this function needs to actually be called in mainState

File: firmware/quadruna/RSM/src/tasks.c [hunk 3]
```diff
@@ -45,52 +87,162 @@ AdcChannel id_to_adc[] = {
...
+        app_stateMachine_tick1Hz();
+
+        const bool debug_mode_enabled = app_canRx_Debug_EnableDebugMode_get();
+        io_canTx_enableMode(CAN_MODE_DEBUG, debug_mode_enabled);
+        io_canTx_enqueue1HzMsgs();
+
+        start_ticks += period_ms;
+        osDelayUntil(start_ticks);
+    }
 }
```
Comment (Lucien950) [inline firmware/quadruna/RSM/src/tasks.c:247]: you need to call `io_coolant_inputCaptureCallback` in a function called `void HAL_TIM_IC_CaptureCallback(TIM_HandleTypeDef *htim)`

**Review by Lucien950 — CHANGES_REQUESTED** [review]

File: firmware/quadruna/RSM/src/app/app_globals.h [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <stdbool.h>
+#include "io_led.h"
+
+typedef struct
+{
+    const BinaryLed *const brake_light;
```
[the remark this answers is not in the document]
Comment (shayana18) [reply firmware/quadruna/RSM/src/app/app_globals.h:8]: I had a discussion with Gus about this. I feel like the big benefit of doing this is just scalability incase we add more leds later or something like the buzzer again. I can still make the change if you think that's best.

**Review by Lucien950 — APPROVED** [review]

**Pull request merged**